R/combine_GVA.R

```diff
@@ -69,123 +71,70 @@ combine_GVA <- function(
-
-  GVA_sectors <- GVA_sectors[which(GVA_sectors$present == TRUE),]
-  GVA_sectors <- dplyr::left_join(GVA_sectors, denom, by = c('year', 'SIC2'))
-  GVA_sectors$perc_split <- GVA_sectors$ABS_ind_GVA / GVA_sectors$ABS_2digit_GVA
-  GVA_sectors <- GVA_sectors[!(is.na(GVA_sectors$year) & is.na(GVA_sectors$ABS_ind_GVA)),]
-
-  # Now, multiply `perc_split` by `GVA` after joining with `GVA` to get the
-  # `BB16_GVA` (column Q in `Working File` worksheet).
-
-  GVA_sectors <- dplyr::left_join(GVA_sectors, GVA, by = c('SIC2' = 'SIC', 'year'))
-  GVA_sectors$BB16_GVA <-  GVA_sectors$perc_split * GVA_sectors$GVA
-
-  ### 4. Calculate sums across sectors and years ----
-
-  # Finally calculate the sums across all sectors and years.
-
-  GVA_by_sector <- dplyr::group_by_(GVA_sectors, ~year, ~sector)
-  GVA_by_sector <- dplyr::summarise_(GVA_by_sector, GVA = ~sum(BB16_GVA))
-
-  #### 5. Add in total UK GVA from gva ----
-
-
-  # Use create a table to merge in with GVA_by_sector
-
-  GVA_UK <- GVA[which(grepl('total.*intermediate.*',GVA$SIC)),]
-  GVA_UK$sector <- 'UK'
-  GVA_UK <- GVA_UK[,c('year', 'sector', 'GVA')]
-
-  # Merge this into `GVA_by_sector` by full join
-
-  GVA_by_sector <- dplyr::full_join(GVA_by_sector, GVA_UK)
-
-  ### 6. Match in Tourism data ----
-
-  # Tourism data is provided in a separate spreadsheet and imported in the
-  # `tourism` object
-
-  tourism_UK <- tourism
-  tourism_UK$sector <- 'tourism'
-  tourism_UK <- tourism_UK[, c('year', 'sector', 'GVA')]
-  GVA_by_sector <- dplyr::full_join(tourism_UK, GVA_by_sector)
-
-  ### 7. Add tourism overlap ----
-
-  # Also need to add the `$overlap` from tourism to the `all_dcms` totals in
-  # `GVA_by_sector`
-
-  tourism_all_sectors <- tourism
-  tourism_all_sectors$sector <- 'all_dcms'
-  tourism_all_sectors <- tourism_all_sectors[ ,c('year', 'sector', 'overlap')]
-  GVA_by_sector <- dplyr::right_join(tourism_all_sectors, GVA_by_sector)
-  GVA_by_sector$GVA <- ifelse(
-    !is.na(GVA_by_sector$overlap),
-    GVA_by_sector$overlap + GVA_by_sector$GVA,
-    GVA_by_sector$GVA
-  )
-  GVA_by_sector <- GVA_by_sector[, !names(GVA_by_sector) == 'overlap']
-
-  #### 8. Build the GVA_by_sector dataframe expected by the eesectors package ----
-
-  # Build df to match eesectors::GVA_by_sector_2016
-  # Will need adjusting in future versions
-
-  GVA_by_sector <- GVA_by_sector[GVA_by_sector$year %in% 2010:2015,]
-  GVA_by_sector <- dplyr::ungroup(GVA_by_sector) # May not be required
-  GVA_by_sector$GVA <- round(GVA_by_sector$GVA, 2)
-  GVA_by_sector$sector <- factor(GVA_by_sector$sector)
-  GVA_by_sector$year <- as.integer(GVA_by_sector$year)
-  GVA_by_sector <- GVA_by_sector[, c('sector', 'year', 'GVA')]
-  GVA_by_sector <- dplyr::arrange_(GVA_by_sector, ~year, ~sector)
 
-  return(GVA_by_sector)
+  #Annual business survey, duplicate 2014 data for 2015 and
+  #then duplicate non SIC91 then add SIC 91 with sales data
+  ABS_2015 <- filter(ABS, year == 2014) %>%
+    mutate(year = 2015) %>%
```

Comment:
Ah looks like this is why you wanted to import dplyr. This is generally considered to be bad practice in package dev. Much better to use `dplyr::mutate()`

Reply:
@ivyleavedtoadflax  as far as I can tell the only downside is that it makes it less clear whether dplyr functions are user written or not. I think the benefits outweigh this - dplyr is clean and easy to read - not so much with dplyr:: on every line

Reply:
There is also a potential namespace clash over `filter`, and possibly other functions too. I defer to the great Hadley here: http://r-pkgs.had.co.nz/namespace.html#imports

> It’s common for packages to be listed in Imports in DESCRIPTION, but not in NAMESPACE. In fact, this is what I recommend: list the package in DESCRIPTION so that it’s installed, then always refer to it explicitly with pkg::fun(). Unless there is a strong reason not to, it’s better to be explicit. It’s a little more work to write, but a lot easier to read when you come back to the code in the future. The converse is not true. Every package mentioned in NAMESPACE must also be present in the Imports or Depends fields.

Reply:
Agree with @ivyleavedtoadflax, I would add the `dplyr::` prefix

Reply:
I disagree, the only objects that are masked by dplyr are: filter, lag, intersect, setdiff, setequal, union. except for filter and lag, the dplyr versions are just wrappers for the base versions. I don't think the potential confusion that might arise if someone tries to use the base versions of filter or lag outweighs the benefit of having clean dplyr code. I think more time will be wasted from missing dplyr:: calls. Remember there is lots more data cleaning to do for eesectors - the package will end up mostly dplyr code - and the dplyr bits are the bits that will need maintaining and updating from publication to publication. That's my two penneth anyway ;)

Reply:
ok. I would just make sure that this will be compatible with DCMS machines. One of the reasons Niall and I stripped out most of the dplyr in the first version was that we simply couldn't get it to run on some of the Windows machines in the department as they had old versions or R and consequently dplyr.

Will you also do this for other packages that you import into eesectors like ggplot:: for consistency?

Reply:
Yes - I will test it on some other laptops in the team - would be useful if you could test on your macbook too. My plan was to only import dplyr - that way namespace clashes can only happen with the base packages. If all other packages functions are called using pckg:: then I don't think they will be any problems. I do need to do a bit more experimenting to double check though.